example/tutorial/launch_bounds/launch_bounds_reduce.cpp

```diff
@@ -38,10 +38,11 @@ struct collision {
   int hash(int q) const {
     // A simple hash by Justin Sobel
     // Thanks to Arash Partow (partow.net)
-    char* fourchars = (char*)&q;
-    int hash        = 1315423911;
-    for (int i = 0; i < 4; fourchars++, i++) {
-      hash ^= ((hash << 5) + *fourchars + (hash >> 2));
+    int hash = 1315423911;
+    for (int i = 0; i < 4; i++) {
+      char c = static_cast<char>(q & 0xff);
+      q >>= 8;
+      hash ^= ((hash << 5) + c + (hash >> 2));
```

Comment:
I checked that this compiles to the same instructions, see https://godbolt.org/z/Ycn6zd6K8.

Reply:
If it ain't broke, don't fix it.  This PR is advertised as adding missing support for building some of the tutorials with CMake.  Let's not let the scope creep.

Reply:
I get the CI error
```
/__w/kokkos/kokkos/example/tutorial/launch_bounds/launch_bounds_reduce.cpp:41:23: error: do not use C-style cast to convert between unrelated types [cppcoreguidelines-pro-type-cstyle-cast,-warnings-as-errors]
   41 |     char* fourchars = (char*)&q;
```

https://github.com/kokkos/kokkos/actions/runs/10185773769/job/28176050168#step:11:828

Can you suggest a better way to fix this?

Reply:
I see.

I would do `// NOLINT(cppcoreguidelines-pro-type-cstyle-cast)`.

Reply:
Sure, I added the NOLINT statement. Let me know in case you are interested in having my proposed change above in a separate PR (this removes the c cast to access the bytes in the int).